pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -1151,14 +1165,25 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
               this.progressTracker.action('container engine components', 50, async() => {
                 await this.writeFile('/etc/containerd/config.toml', CONTAINERD_CONFIG);
                 await this.writeConf('containerd', { log_owner: 'root' });
-                await this.writeFile('/etc/init.d/docker', SERVICE_SCRIPT_DOCKERD, { permissions: 0o755 });
+                await this.writeFile('/usr/local/bin/nerdctl', NERDCTL, 0o755);
+                await this.writeFile('/etc/init.d/docker', SERVICE_SCRIPT_DOCKERD, 0o755);
                 await this.writeConf('docker', {
                   WSL_HELPER_BINARY: await this.getWSLHelperPath(),
                   LOG_DIR:           logPath,
                 });
-                await this.writeFile(`/etc/init.d/buildkitd`, SERVICE_BUILDKITD_INIT, { permissions: 0o755 });
+                await this.writeFile(`/etc/init.d/buildkitd`, SERVICE_BUILDKITD_INIT, 0o755);
                 await this.writeFile(`/etc/conf.d/buildkitd`, SERVICE_BUILDKITD_CONF);
               }),
+              this.progressTracker.action('Openresty configuration', 50, async() => {
+                const allowListConf = BackendHelper.createImageAllowListConf(config.containerEngine.imageAllowList);
+                const resolver = `resolver ${ await this.ipAddress } ipv6=off;\n`;
```

Comment:
For clarification for myself:
This goes `openresty` → (WSL VM eth0) → host-resolver → (af_vsock) → WSL host machine.
The host-resolver part may be replaced by dnsmasq depending on configuration, in which case it talks to the WSL built-in DNS server as upstream. We might remove this soon.

Reply:
> This goes openresty → (WSL VM eth0) → host-resolver → (af_vsock) → WSL host machine.

That sounds correct.

> The host-resolver part may be replaced by dnsmasq depending on configuration,

When does this happen? But wouldn't the logic still be correct? We want to connect to the nameserver from `/etc/resolv.conf`, and I thought that one always points to the eth0 IP address.

Reply:
We still have a `hostResolver` configuration, though I don't expect that to have any use at this point.
https://github.com/rancher-sandbox/rancher-desktop/blob/e1853c803182fd5c1c076eb3b6c259200adb8cc7/pkg/rancher-desktop/config/settings.ts#L63-L67